docs/core/cpi.md

```diff
@@ -1,360 +1,100 @@
-implemented with `Pubkey::create_with_seed`.
-
-For reference, that implementation is as follows:
+The
+[`invoke`](https://github.com/solana-labs/solana/blob/master/sdk/program/src/program.rs#L132)
+function is used when making a CPI that does not require PDA signers. When
+making CPIs, signers provided to the caller program can extend to the callee
+program.
 
 ```rust
-pub fn create_with_seed(
-    base: &Pubkey,
-    seed: &str,
-    program_id: &Pubkey,
-) -> Result<Pubkey, SystemError> {
-    if seed.len() > MAX_ADDRESS_SEED_LEN {
-        return Err(SystemError::MaxSeedLengthExceeded);
-    }
-
-    Ok(Pubkey::new(
-        hashv(&[base.as_ref(), seed.as_ref(), program_id.as_ref()]).as_ref(),
-    ))
-}
+pub fn invoke(
+    instruction: &Instruction,
+    account_infos: &[AccountInfo<'_>]
+) -> Result<(), ProgramError>
 ```
 
-Programs can deterministically derive any number of addresses by using seeds.
-These seeds can symbolically identify how the addresses are used.
-
-From `Pubkey`::
-
-```rust
-/// Generate a derived program address
-///     * seeds, symbolic keywords used to derive the key
-///     * program_id, program that the address is derived for
-pub fn create_program_address(
-    seeds: &[&[u8]],
-    program_id: &Pubkey,
-) -> Result<Pubkey, PubkeyError>
+Here is an example program on
+[Solana Playground](https://beta.solpg.io/github.com/ZYJLiu/doc-examples/tree/main/cpi-invoke)
+that makes a CPI using the `invoke` function to call the transfer instruction on
+the System Program.
 
-/// Find a valid off-curve derived program address and its bump seed
-///     * seeds, symbolic keywords used to derive the key
-///     * program_id, program that the address is derived for
-pub fn find_program_address(
-    seeds: &[&[u8]],
-    program_id: &Pubkey,
-) -> Option<(Pubkey, u8)> {
-    let mut bump_seed = [std::u8::MAX];
-    for _ in 0..std::u8::MAX {
-        let mut seeds_with_bump = seeds.to_vec();
-        seeds_with_bump.push(&bump_seed);
-        if let Ok(address) = create_program_address(&seeds_with_bump, program_id) {
-            return Some((address, bump_seed[0]));
-        }
-        bump_seed[0] -= 1;
-    }
-    None
-}
-```
+You can also reference the [CPI Basic Example](/docs/core/cpi/example1) page for
+further details.
 
-**Warning**: Because of the way the seeds are hashed there is a potential for
-program address collisions for the same program id. The seeds are hashed
-sequentially which means that seeds {"abcdef"}, {"abc", "def"}, and {"ab", "cd",
-"ef"} will all result in the same program address given the same program id.
-Since the chance of collision is local to a given program id, the developer of
-that program must take care to choose seeds that do not collide with each other.
-For seed schemes that are susceptible to this type of hash collision, a common
-remedy is to insert separators between seeds, e.g. transforming {"abc", "def"}
-into {"abc", "-", "def"}.
+### CPI with PDA Signer
```

Comment:
Could you go into more of why you would want a signer here? https://solana.com/docs/core/cpi#instructions-that-require-privileges goes pretty in depth into it, especially with the common use case of a program owning a token account